lib/appmap/service/validator/config_validator.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'appmap/service/validator/violation'
+
+module AppMap
+  module Service
+    module Validator
+      class ConfigValidator
+        attr_reader :config, :violations
+
+        def initialize(config_file)
+          @config_file = config_file
+          @violations = []
+          @config = load_config
+        end
+
+        def valid?
+          validate_config_presence
+          @violations.length == 0
+        end
+
+        private
+
+        def present?
+          File.exist?(@config_file)
+        end
+
+        def load_config
+          AppMap::Config.load_from_file(@config_file) if present?
+        rescue StandardError => e
+          @violations << Violation.error(
+            filename: @config_file,
+            message: 'AppMap configuration is not valid YAML',
```

Comment:
The config file might be valid YAML here, but some other error occurred. That's why I suggest splitting this in two.